### PR TITLE
moderation: add note system

### PR DESCRIPTION
**Pull request by l-zeuch**
This code is fairly old and I haven't touched it in ages, so it
definitely needs some testing.
However, I did touch up a few lines here and there, to be more up to
date.

Additionally, I decided against a complete refactor of the storage,
as it was my intention to do it like that. This sentiment is also
reflected in the `Trivia` section of the documentation.

Signed-off-by: Luca Zeuch <l-zeuch@email.de>

**Status**

- [ ] Code changes have been tested on an instance of YAGPDB, or there are no code changes
- [x] I have read and followed the [contribution guide](https://github.com/yagpdb-cc/yagpdb-cc/blob/master/CONTRIBUTING.md)
- [x] I have [written documentation](https://github.com/yagpdb-cc/yagpdb-cc/blob/master/WRITING-DOCUMENTATION.md) for my changes, or there is no need to

### CODE REVIEW

File: src/moderation/notes/notes_minified.go.tmpl [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
...il 10 "exec"}}{{end}}{{else}}{{$err = "Wrong password. Run the `nuke` command again to generate a new one."}}{{end}}{{end}}{{else}}{{$err = printf "You silly, twisted boy, you. (Missing permissions)"}}{{end}}{{else}}{{$err = printf "`%s` is not a valid subcommand!\nValid subcommands are: `%s`.\nRun `%snotes help` for more information." $subcommand $subcommands_string $prefix}}{{end}}{{else}}{{$err = "Member not found."}}{{end}}{{end}}{{else}}{{$err = printf "Just what do you think you're doing Dave? (Missing permissions)" }}{{end}}{{if $err}}{{$err = cembed "description" $err "author" (sdict "name" "An Error Occurred:") "color" 0xff0000"footer" (sdict "text" (printf "Triggered by: %s" .User) "icon_url" (.User.AvatarURL "1024"))}}{{sendMessage nil $err}}{{end}}{{if and $out (not $err)}}{{sendMessage nil (cembed $out)}}{{end}}{{else}}{{$count := dbCount "notes"}}{{if gt $count 0}}{{if ge $count 100}}{{$dump := dbDelMultiple (sdict "pattern" "notes") 100 0}}{{execCC .CCID nil 10 "exec"}}{{else}}{{$dump := dbDelMultiple (sdict "pattern" "notes") $count 0}}{{execCC .CCID nil 10 "exec"}}{{end}}{{else}}{{sendMessage nil (cembed "title" "Successfully purged entire system.")}}{{end}}{{end}}
```
Comment (jo3-l) [inline src/moderation/notes/notes_minified.go.tmpl:12]: CI is failing with
```
unexpected "\"fields\"" in operand
```

Though the action might just be broken, have you checked that this saves and runs properly?

**Review by jo3-l — COMMENTED** [review]
Thanks for this. One comment:

File: src/moderation/notes/notes_minified.go.tmpl (hunk 1 above)
Comment (l-zeuch) [reply src/moderation/notes/notes_minified.go.tmpl:12]: Hm, that's interesting. I'll check quickly, all I did was to see if it's below 10000 characters :^)

Comment (l-zeuch) [reply src/moderation/notes/notes_minified.go.tmpl]: Joe, if you don't mind -- could you run the minifier on that code? It doesn't work for me (perhaps due to my aggressive blocking, I don't want to troubleshoot that right now).

Knowing that you know how to config your minifier, that seems the best shot. There's a plethora of complaints yagpdb is throwing at me and it's not fun to parse through minified code.

The normal large code is fine though.

Comment (jo3-l) [reply src/moderation/notes/notes_minified.go.tmpl]: I can take a look tomorrow.

Comment (l-zeuch) [reply src/moderation/notes/notes_minified.go.tmpl]: Thanks! Appreciate it.

Comment (l-zeuch) [reply src/moderation/notes/notes_minified.go.tmpl]: Should be good now -- a sneaky `)` sneaked in.

**Review by jo3-l — CHANGES_REQUESTED** [review]
Please restructure the website page to conform with [the template](https://github.com/yagpdb-cc/yagpdb-cc/blob/master/website/TEMPLATE.md) more closely, for consistency.

File: website/docs/moderation/notes.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+<AvailablePermissionsDisplay />
+
+- 📌 $BASE_PERMISSION <br />
+This permission should be the permission which all server staff have - on most servers, this is `ManageMessages`. If you want to lock this system behind a more exclusive permission, please view the full list under `$NUKE_PERMISSION` - it lists all of them.
+
+- 📌 $NUKE_PERMISSION <br />
+  This permission is quite a dangerous one: It grants the ability to purge the entire system. Most servers consider users with `ManageServer` permission as an admin, hence this is the default. It is recommended to change this to a more exclusive permission rather than a common one. `Adminstrator` would be a fitting candidate, as it's a quite rarely given permission, `ManageRoles` however may not.
+
+- 📌 $DELETE*TIMEOUT <br />
```
Comment (l-zeuch) [inline website/docs/moderation/notes.md:55]: Somehow this displays as a `*`, but it's an underscore as intended in my editor >
image

**Review by jo3-l — APPROVED** [review]
LGTM

**Pull request merged**